public/app/features/explore/ShortLinkButtonMenu.tsx

```diff
@@ -0,0 +1,99 @@
+import { css } from '@emotion/css';
+import React, { useState } from 'react';
+
+import { GrafanaTheme2, urlUtil } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import { reportInteraction } from '@grafana/runtime';
+import { useStyles2, ToolbarButton, Dropdown, Menu } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+import { createAndCopyShortLink } from 'app/core/utils/shortLinks';
+import { useSelector } from 'app/types';
+
+import { getUrlStateFromPaneState } from './hooks/useStateSync';
+import { selectPanes } from './state/selectors';
+
+export function ShortLinkButtonMenu() {
+  const styles = useStyles2(getStyles);
+  const panes = useSelector(selectPanes);
+  const [isOpen, setIsOpen] = useState(false);
+  const onCopyShortLink = (url?: string) => {
+    createAndCopyShortLink(url || global.location.href);
+    reportInteraction('grafana_explore_shortened_link_clicked');
+  };
+
+  const MenuActions = () => {
+    return (
+      <Menu>
+        <Menu.Item
+          key="copy-link"
+          label="Copy shortened URL"
```

Comment:
maybe we can use the same phrasing as the other button for consistency? (also to avoid confusion if someone wonders what is the difference between a "shortened link" and a "shortened URL")
```suggestion
          label={t('explore.toolbar.copy-shortened-link', 'Copy shortened link')}
```

Reply:
fixed!